### PR TITLE
chore: Upgrade `revm`

### LINKED ISSUE #5490: Upgrade `revm` in `reth` to a version including the `Canyon` hardfork

### PR DESCRIPTION
**Pull request by clabby**
## Overview

> **Note**
> Do not merge, currently patched with my own fork. Will redirect at `revm`'s head once https://github.com/bluealloy/revm/pull/871 has been merged.

Upgrades `revm` to the latest version, which includes the `Canyon` hardfork. This unblocks implementing the `Canyon` hardfork for `op-reth`, tracked in #5210.

**Metadata**
Closes #5490

### CODE REVIEW

File: crates/revm/revm-inspectors/src/stack/maybe_owned.rs [hunk 1]
```diff
@@ -85,86 +90,92 @@ where
...
         }
 
-        (InstructionResult::Continue, Gas::new(0), Bytes::new())
+        Some((
+            InterpreterResult {
+                result: InstructionResult::Continue,
+                gas: Gas::new(0),
+                output: Bytes::new(),
+            },
+            0..0,
```
Comment (clabby) [inline crates/revm/revm-inspectors/src/stack/maybe_owned.rs:129]: @rakita What does this new range in the result here signify? Tests are passing, though I don't think this is correct. Occurs in several places.

**Comment by clabby** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5489** <a href="https://app.graphite.dev/github/pr/paradigmxyz/reth/5489?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/paradigmxyz/reth/5489?utm_source=stack-comment).

File: crates/revm/revm-inspectors/src/stack/maybe_owned.rs (hunk 1 above)
Comment (rakita) [reply crates/revm/revm-inspectors/src/stack/maybe_owned.rs:129]: If you want to shortcircuit the call you can return the result. So this should be `None`

File: crates/revm/revm-inspectors/src/stack/maybe_owned.rs [hunk 2]
```diff
@@ -85,86 +90,92 @@ where
...
 
-        (InstructionResult::Continue, None, Gas::new(0), Bytes::default())
+        Some((
+            InterpreterResult {
+                result: InstructionResult::Continue,
+                gas: Gas::new(0),
+                output: Bytes::new(),
+            },
+            None,
+        ))
```
Comment (rakita) [inline crates/revm/revm-inspectors/src/stack/maybe_owned.rs:162]: Same here, it should be None as in here: https://github.com/paradigmxyz/reth/pull/5489/files/880fd26ca13093810afc7f6d2e1e9617b44c0ecb#r1398242606

**Pull request closed without merging**